rust/connlib/tunnel/src/tests/transition.rs

```diff
@@ -339,13 +339,17 @@ fn dns_transport() -> impl Strategy<Value = DnsTransport> {
     prop_oneof![Just(DnsTransport::Udp), Just(DnsTransport::Tcp),]
 }
 
-pub(crate) fn query_type() -> impl Strategy<Value = Rtype> {
-    prop_oneof![
-        Just(Rtype::A),
-        Just(Rtype::AAAA),
-        Just(Rtype::MX),
-        Just(Rtype::PTR),
-    ]
+pub(crate) fn query_type(mut rtypes: Vec<Rtype>) -> impl Strategy<Value = Rtype> {
+    if rtypes.contains(&Rtype::A) || rtypes.contains(&Rtype::AAAA) {
```

Comment:
What is the thinking behind this logic?

Reply:
We never really add `MX` records in the test but I didn't want to remove that transition.

And to be able to test nat46 we need to query for A when only AAAA is available and vice-versa so I always query for both.

Reply:
Similar thing for `PTR`, for it to make sense we need an `A` or `AAAA` record.

Reply:
So the variable that we get passed are the available record types? Can we rename a few things here to make that more obvious? And perhaps not use mutation?

Reply:
> We never really add `MX` records in the test but I didn't want to remove that transition.
> 
> And to be able to test nat46 we need to query for A when only AAAA is available and vice-versa so I always query for both when one is available.

Do you mind documenting this on the function?